From 6fd539a2559076239eb09a3a49b7ff7ca53ea00e Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:58 -0400
Subject: [misc] mei: bus: propagate error code returned by mei_me_cl_by_id

Message-id: <1399041470-21330-26-git-send-email-prarit@redhat.com>
Patchwork-id: 79312
O-Subject: [RHEL7.1 PATCH BZ 1088953 25/77] mei: bus: propagate error code returned by mei_me_cl_by_id
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit f9350129a09d8f55ba9322fa4327aeb499a91bbd
Author: Tomas Winkler <tomas.winkler@intel.com>
Date:   Mon Oct 21 22:05:40 2013 +0300

    mei: bus: propagate error code returned by mei_me_cl_by_id

    no need to change error code value returned by
    mei_me_cl_by_id, just propagate it on

    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/bus.c b/drivers/misc/mei/bus.c
index cd2033c..4bc7d62 100644
--- a/drivers/misc/mei/bus.c
+++ b/drivers/misc/mei/bus.c
@@ -245,7 +245,7 @@ static int ___mei_cl_send(struct mei_cl *cl, u8 *buf, size_t length,
  /* Check if we have an ME client device */
  id = mei_me_cl_by_id(dev, cl->me_client_id);
  if (id < 0)
-  return -ENODEV;
+  return id;
 
  if (length > dev->me_clients[id].props.max_msg_length)
   return -EINVAL;
-- 
1.7.1